backend/rxnorm_normalize.py

```
@@ -0,0 +1,564 @@
+
+def _input_key(name: str) -> str:
+    return name.strip().lower()
+
+
+def _cache_lookup(conn: sqlite3.Connection, key: str) -> Optional[Dict]:
+    cursor = conn.execute(
+        f"""
+        SELECT rxcui, normalized_name, ingredient_rxcui, ingredient_name, matched
+        FROM {CACHE_TABLE}
+        WHERE input_key = ?
+        """,
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)